### DISCUSSION TITLE
How do I get my client to support TLS 1.0?

### DISCUSSION THREAD

**Discussion by ONiKiDSK** [baresip/baresip#2968] [Q&A]
(baresip 3.10.0 , openssl 3.1.0)I receive this error when connecting to my voip provider:|
  tls: connect: error (r=-1, ssl_err=1)
  tls: B07C3E7A6E000000:error:0A000102:SSL routines:ssl_choose_client_version
  unsupported protocol:ssl/statem/statem_lib.c:1947:
  if (SSL_IS_DTLS(s) ? DTLS_VERSION_LT(s->version, ver_min)
                       : s->version < ver_min) {
        s->version = origv;
        SSLfatal(s, SSL_AD_PROTOCOL_VERSION, SSL_R_UNSUPPORTED_PROTOCOL);
        return 0;
    }
[statem_lib.c.txt](https://github.com/baresip/baresip/files/14613671/statem_lib.c.txt)

Because our server only supports TLSv1.0, and my client does not support TLSv1.0, how can I change it so that the client supports TLSv1.0?
Please grant me this unreasonable request

**Comment by alfredh** [comment]
Please note that TLS 1.0 has been deprecated:
https://www.rfc-editor.org/rfc/rfc8996

Please post these types of questions in the "Discussions".

**Comment by robert-scheck** [comment]
This is most likely less an issue of Baresip, but more related to the OpenSSL configuration and/or the OpenSSL defaults your (Linux) distribution ships (e.g. Debian disabled TLSv1.0 in OpenSSL by default with Debian 10). To provide a more helpful answer, you would have to provide information about the (Linux) distribution, you are using.
